core/src/main/kotlin/live/LiveKordEntity.kt

```diff
@@ -44,8 +50,23 @@ abstract class AbstractLiveKordEntity(dispatcher: CoroutineDispatcher) : LiveKor
     protected abstract fun filter(event: Event): Boolean
     protected abstract fun update(event: Event)
 
-    override fun shutDown() {
-        cancel("Shutdown executed")
+    fun onShutdown(action: (() -> Unit)?) {
+        val currentAction = shutdownAction
+
+        shutdownAction =
+            if (action == null) {
+                currentAction?.second?.cancel()
+                null
+            } else {
+                action to (currentAction?.second ?: on<Event> {})
+            }
+    }
```

Comment:
By the way of `coroutineContext.job.invokeOnCompletion(block)` ?

Reply:
I don't think you can get the supervisor from the context directly (correct me if I'm wrong). If that's correct you'll have to keep a reference to the supervisor job in the entity for that to work.

Reply:
I just tested and it works, so we can remove all system about the shutdown action, and just use `live.coroutineContext.job.invokeOnCompletion {}`. What do you think about it ?

Reply:
Less code sounds better 👍

Reply:
If we use the method invokeOnCompletion, we don't need to save anything in the entity (even an event as said previously) .. so that solves the problem !